inspirehep/modules/converttohal/tei.py

```diff
@@ -0,0 +1,131 @@
+from jinja2 import Environment, PackageLoader
+from nameparser import HumanName
+
+from inspirehep.modules.records.json_ref_loader import replace_refs
+from inspirehep.utils.dedupers import dedupe_list
+from inspirehep.utils.record import get_value
+
+TEMPLATE = 'default.xml'
+
+
+def tei_response(record):
+    """Returns the record formatted in XML+TEI per HAL's specification.
+
+    Keyword arguments:
+    record -- the record to convert
+    """
+    env = Environment(loader=PackageLoader('inspirehep.modules.converttohal',
+                                           'templates'),
+                      trim_blocks=True, lstrip_blocks=True)
+    template = env.get_template(TEMPLATE)
+
+    authors = _parse_authors(record)
+
+    titles = record.get('titles', [])
+
+    doi = _parse_dois(record)
+
+    publication = _parse_pub_info(record)
+
+    structures = _parse_structures(record)
+
+    return template.render(titles=titles, doi=doi, authors=authors,
+                           publication=publication, structures=structures)
+
+
+def _parse_structures(record):
+    structures = []
+    for author in record.get('authors', []):
+        for affiliation in author.get('affiliations', []):
+            ref = replace_refs(affiliation, 'db')
+            try:
+                if ref['record']:
+                    structures.append(
+                        _structure_data(ref['record'], ref['recid'])
+                    )
+            except KeyError:
+                continue
+    return dedupe_list(structures)
+
+
+def _parse_pub_info(record):
+    try:
+        pub_info = _lookup(record, "publication_info[0]")
+        if 'journal_title' in pub_info:
+            return _journal_data(pub_info)
+        elif 'conference_record' in pub_info:
+            return _conference_data(pub_info['conference_record'])
+        else:
+            return None
+    except KeyError:
+        return None
+
+
+def _parse_dois(record):
+    return get_value(record, "dois[0].value", "")
+
+
+def _parse_authors(record):
+    # FIXME: add HAL ID to authors
+    authors = get_value(record, "authors", [])
+    for author in authors:
+        try:
+            parsed = HumanName(author['full_name'])
+            author['parsed_name'] = parsed
+        except KeyError:
+            continue
+    return authors
+
+
+def _conference_data(conf):
+    ref = replace_refs(conf, 'db')
+
+    # FIXME: Add conference city, country, and country code fields
+    if ref:
+        return {'type': "conference",
+                'name': get_value(ref, "titles[0].title", ""),
+                'acronym': get_value(ref, "acronym[0]", ""),
+                'opening_date': get_value(ref, "opening_date", ""),
+                'closing_date': get_value(ref, "closing_date", "")}
+    else:
+        return {'type': "conference",
+                'name': "",
+                'acronym': "",
+                'opening_date': "",
+                'closing_date': ""}
+
+
+def _journal_data(jour):
```

Comment:
Can you call it `pub_info` rather than `jour` (which is the French for `day` :smile: )